app/src/main/java/rick_and_morty/ui/characters/CharactersViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package rick_and_morty.ui.characters
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import rick_and_morty.data.model.CharacterResultsDto
+import rick_and_morty.data.repository.CharacterRepository
+import javax.inject.Inject
+
+@HiltViewModel
+class CharactersViewModel @Inject constructor(
+    private val characterRepository: CharacterRepository
+) : ViewModel(){
+
+    var initialLoad: Boolean = true
+
+    private var _characters = MutableStateFlow<List<CharacterResultsDto>>(emptyList())
+    val characters = _characters.asStateFlow()
+
+    fun getCharacters(page: Int) {
+        viewModelScope.launch {
+            characterRepository.getCharacters(page)
```

Comment:
this looks like it will crash if it fails, e.g. no internet....

Reply:
Try and catch

Reply:
Done